modules/apps/commerce/commerce-product-content-web/src/main/java/com/liferay/commerce/product/content/web/internal/info/item/renderer/OptionSelectorInfoItemRenderer.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.product.content.web.internal.info.item.renderer;
+
+import com.liferay.commerce.account.model.CommerceAccount;
+import com.liferay.commerce.constants.CommerceWebKeys;
+import com.liferay.commerce.context.CommerceContext;
+import com.liferay.commerce.product.content.constants.CPContentWebKeys;
+import com.liferay.commerce.product.content.util.CPContentHelper;
+import com.liferay.commerce.product.model.CPDefinition;
+import com.liferay.info.item.renderer.InfoItemRenderer;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.theme.PortletDisplay;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.util.Locale;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Alessio Antonio Rendina
+ */
+@Component(
+	service = {InfoItemRenderer.class, OptionSelectorInfoItemRenderer.class}
+)
+public class OptionSelectorInfoItemRenderer
+	implements InfoItemRenderer<CPDefinition> {
+
+	@Override
+	public String getKey() {
+		return "cpDefinition-option-selector";
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		return _language.get(locale, "option-selector");
+	}
+
+	@Override
+	public void render(
+		CPDefinition cpDefinition, HttpServletRequest httpServletRequest,
+		HttpServletResponse httpServletResponse) {
+
+		if (cpDefinition == null) {
+			return;
+		}
+
+		try {
+			CommerceContext commerceContext =
```

Comment:
what if you put this fragment outside of Commerce?

Reply:
This fragment can be used only for CPDefinition

Reply:
`implements InfoItemRenderer<CPDefinition>`

Reply:
got it, right! now I see, thank you!